File: 1.1 BinaryGap.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public static int solution(int N) {
        if (N>1){
            int2binary(N);
            int i = binaryString.length();
            counter = 0;
            result = 0;
            for (int j=1;j<i;j++){
                if (binaryString.charAt(j)=='0'){
                counter = counter + 1;
            }
                else if(binaryString.charAt(j)=='1'){
                    if(counter>result){
                        result = counter;
                    }
                    counter = 0;
                }
            }
        }
        else{
            result = 0;
        }
        return result;
    }

    public static String binaryString; 
    public static int result; 
    public static int counter;

    public static String int2binary(int n) {
        binaryString = Integer.toBinaryString(n);
        return binaryString;
    }
}

File: 1. Iterations/1.1 BinaryGap.go
package solution

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

import (
	"strconv"
)

func Solution(N int) int {
	binary := strconv.FormatInt(int64(N), 2)
	maxGap := 0
	currentGap := 0
	for _, char := range binary {
		if char == '0' {
			currentGap++
		} else {
			if currentGap > maxGap {
				maxGap = currentGap
			}
			currentGap = 0
		}
	}
	return maxGap
}
